### PR TITLE
Help form designers test+stop testing

### PR DESCRIPTION
**Pull request by samuelhwilliams**
Good for first review/feedback/"do we want this" - needs tests/tweaking for final review

## 🎫 Ticket
BAU

## 📝 Description
Adds a button on individual 'manage form/task' pages so that form designers can jump straight into testing an individual form. This is easy while we have no cross-form interactions; we'll have to think through how we might facilitate this in a world where there are dependencies _between_ forms.

It also increases the height+visibility of the 'testing banner', and adds a button to any of the form runner pages so that a form designer can bail out of the testing journey. The theory is that a form designer is looking to test a very specific behaviour, and as soon as they've done that they'll want to get back to tweaking/editing the form configuration itself. We shouldn't force them to do an awkward multi-back nav, or complete an entire form/submission.

Note: the flash message when testing is 'complete' don't feel very nice. I feel like you need _something_ to explain the unusual navigation, but can't think of something that looks/feels good, so this is very much just a bad attempt.

### Review Readiness
- [x] PR title and description provide sufficient context for reviewers
- [x] Commits are logical, self-contained, and have clear descriptions

### Testing
- [x] I have tested this change and it meets the acceptance criteria for the ticket
- [x] I need the reviewer(s) to pull and run this change locally
- [-] New (non-developer) functionality has appropriate unit and integration tests
- [-] End-to-end tests have been updated (if applicable)
- [-] Edge cases and error conditions are tested

### CODE REVIEW

**Review by srh-sloan — APPROVED** [review]
Looks good, and works when I pulled it locally.
It's so nice we can do things like that and impact every grant/form easily, much better than the form runner! 🙌

File: app/assets/main.scss [hunk 1]
```diff
@@ -157,19 +157,17 @@ $app-destructive-link-active-colour: $govuk-text-colour;
 }
 
 .app-test-data-banner {
+    position: relative;
     text-align: center;
-    border-top: 5px solid #f47738;
-    z-index: 10;
-    margin-bottom: -29px !important;  // 29px hardcoded sorry; dont think we can calculate this
+    background-color: #f47738;
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply app/assets/main.scss:162]: good thought - maybe we can use `govuk-colour("orange")`

File: app/developers/helpers.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    collection: "Collection", return_url: str, form: Optional["Form"] = None
+) -> ResponseReturnValue:
+    user = interfaces.user.get_current_user()
+    delete_submissions_created_by_user(grant_id=collection.grant_id, created_by_id=user.id)
+    submission = create_submission(collection=collection, created_by=user, mode=SubmissionModeEnum.TEST)
+    helper = SubmissionHelper(submission)
+
+    # relies on the return_url not having any query params
+    if "?" in return_url:
+        raise RuntimeError("do better")
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply app/developers/helpers.py]: removed from refactoring

File: app/developers/helpers.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    # we should probably make the session some kind of pydantic model so that we have all the possible stuff in
+    # one place
+    session["test_submission_return_url"] = return_url + "?testing"
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply app/developers/helpers.py]: refactored, good idea

**Review by sfount — APPROVED** [review]
